src/vec/FixedPairList.cpp

```diff
@@ -0,0 +1,345 @@
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  ESPResSo++ is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "FixedPairList.hpp"
+#include "vec/storage/StorageVec.hpp"
+
+#include "python.hpp"
+#include "storage/Storage.hpp"
+#include "boost/bind.hpp"
+#include "boost/serialization/vector.hpp"
+#include "esutil/Error.hpp"
+#include "Buffer.hpp"
+
+#include <sstream>
+
+using namespace std;
+
+namespace espressopp { namespace vec {
+
+  LOG4ESPP_LOGGER(FixedPairList::theLogger, "FixedPairList");
+
+  FixedPairList::FixedPairList(shared_ptr< espressopp::storage::Storage > storage)
+    : globalPairs()
+  {
+    LOG4ESPP_INFO(theLogger, "construct FixedPairList");
+
+    if(!storage->getSystem()->vectorization) {
+      throw std::runtime_error("system has no vectorization");
+    }
+    vectorization = storage->getSystem()->vectorization;
+
+    if(!(vectorization->storageVec))
+      throw std::runtime_error("vectorization->storageVec cannot be null");
+    auto& storageVec = vectorization->storageVec;
+
+    sigBeforeSend = storage->beforeSendParticles.connect(
+      boost::bind(&FixedPairList::beforeSendParticles, this, _1, _2));
+    sigAfterRecv = storage->afterRecvParticles.connect(
+      boost::bind(&FixedPairList::afterRecvParticles, this, _1, _2));
+    sigOnParticlesChanged = storage->onParticlesChanged.connect(
+      boost::bind(&FixedPairList::onParticlesChanged, this));
+
+    if(vectorization->getVecLevel() != 2) {
+      throw std::runtime_error("espressopp::vec::FixedPairList can only be used for vecLevel=2");
+    }
+  }
+
+  FixedPairList::~FixedPairList()
+  {
+    LOG4ESPP_INFO(theLogger, "~FixedPairList");
+    sigBeforeSend.disconnect();
+    sigAfterRecv.disconnect();
+    sigOnParticlesChanged.disconnect();
+  }
+
+  real FixedPairList::getLongtimeMaxBondSqr()
+  {
+    return longtimeMaxBondSqr;
+  }
+
+  void FixedPairList::setLongtimeMaxBondSqr(real d)
+  {
+    longtimeMaxBondSqr = d;
+  }
+
+  void FixedPairList::resetLongtimeMaxBondSqr()
+  {
+    longtimeMaxBondSqr = 0.0;
+  }
+
+  bool FixedPairList::add(size_t pid1, size_t pid2)
+  {
+    bool returnVal = true;
+    if (pid1 > pid2)
+      std::swap(pid1, pid2);
+
+    auto const& storageVec = vectorization->storageVec;
+    size_t const p1 = storageVec->lookupRealParticleVec(pid1);
+    size_t const p2 = storageVec->lookupLocalParticleVec(pid2);
+
+    if(p1==VEC_PARTICLE_NOT_FOUND) {
+      // Particle does not exist here, return false
+      returnVal=false;
```

Comment:
```suggestion
      return false;
```

Makes following `if` unnecessary.

Reply:
This is needed to trigger LOG4ESPP_DEBUG before return as implemented in original version.

Reply:
Why not move the `LOG4ESPP_DEBUG` to this place as well?

Reply:
Right, and I've also removed the returnVal variable entirely